packages/nimble/R/genCpp_sizeProcessing.R

```diff
@@ -2560,11 +2560,13 @@ sizeReturn <- function(code, symTab, typeEnv) {
     asserts <- recurseSetSizes(code, symTab, typeEnv)
     if(inherits(code$args[[1]], 'exprClass')) {
         if(!code$args[[1]]$isName) {
-##            if(code$args[[1]]$toEigenize == 'yes') {
-            if(anyNonScalar(code$args[[1]])) {
+            liftArg <- FALSE
+            if(code$args[[1]]$toEigenize == 'yes')
```

Comment:
There are two aspects to your question.  Yes, if `code$args[[1]]$toEigenize == 'maybe'`, the code will not be eigenized.  The canonical case would be `A + B` where both are scalars.  This is marked `maybe` because if it is part of `C + A + B` where `C` is a vector, then it is fine to include the scalar arithmetic along with the vector arithmetic.  But otherwise the `maybe` would not be eigenized because it can be handled without eigen.  `maybe` can be interpreted as "either".  

Regarding propagation of `toEigenize`: currently each size handler needs to determine if there are any conflicts and create an intermediate variable if necessary.  Example: if `foo` is an RCfunction and is used as `foo(A %*% b)`, the size handler for `foo` (which will be `generalFunSizeHandler`) is responsible for creating an intermediate for `A %*% b`.  

This is an example of what we've chatted about: some steps of eigenization are enmeshed in the type (size) processing, while some are cleanly separated.  Determination of which operations can be eigenized is enmeshed, while transformation of the syntax tree for eigen syntax is separate.  We could benefit from separating some of the enmeshed steps.

I'd like to merge this PR because I plan to modify `sizeReturn` by adding error-trapping on incorrect types.